### ISSUE TITLE
osquery wodle on Windows ERROR: Couldn't load configuration file

### ISSUE THREAD

**Issue by MaxDiOrio** [wazuh/wazuh#21671]
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 4.7.2 | Wazuh component | Agent | Packages/ | Windows 2012 R2 |

Using a centralized config for Windows agents.  Have osquery deployed - osquery.conf works fine on its own.

I want to use the centralized config for the group to push the osquery wodle config including packs.

```
<agent_config>
	<!-- osquery -->
	<wodle name="osquery">
		<disabled>no</disabled>
		<run_daemon>yes</run_daemon>
		<bin_path>C:\Program Files\osquery\osqueryd</bin_path>
		<log_path>C:\Program Files\osquery\log\osqueryd.results.log</log_path>
		<config_path>C:\Program Files\osquery\osquery.conf</config_path>
		<add_labels>no</add_labels>
		<pack name="osquery-monitoring">c:\program files\osquery\packs\osquery-monitoring.conf</pack>
	</wodle>
</agent_config>
```

As soon as I put a pack name in there, I get: 
`wazuh-modulesd:osquery: ERROR: Couldn't load configuration file 'C:\Program Files\osquery\osquery.conf'. Maybe format is invalid.`

I've also tried using an escaped path c:\\program files\\osquery\\packs\\osquery-monitoring.conf with the same error.

For the osquery.conf on the client - I had to escape the path for there to be no errors when using a centralized config without packs.

osquery.conf
```
{
  "options": {
    "utc": "true",
    "disable_events": "true"
  },
  "schedule": {
    "windows_services": {
    "query": "select name,service_type,display_name,pid,start_type,path,module_path,description,user_account from services where status='RUNNING';",
    "interval": 600
    },
    "users": {
      "query": "select * from users;",
      "interval": 600
    }
  },
  "packs": {
    "osquery-monitoring": "c:\\program files\\osquery\\packs\\osquery-monitoring.conf",
    "incident-response": "c:\\program files\\osquery\\packs\\incident-response.conf",
    "it-compliance": "c:\\program files\\osquery\\packs\\it-compliance.conf",
    "vuln-management": "c:\\program files\\osquery\\packs\\vuln-management.conf",
    "hardware-monitoring": "c:\\program files\\osquery\\packs\\hardware-monitoring.conf",
    "ossec-rootkit": "c:\\program files\\osquery\\packs\\ossec-rootkit.conf",
    "windows-attacks": "c:\\program files\\osquery\\packs\\windows-attacks.conf",
    "windows-hardening": "c:\\program files\\osquery\\packs\\windows-hardening.conf"
    }
}
```

I tried pushing the pack file using centralized config as well, and while the files get sent to the agent, adding a pack line into the wodle config fails with the same error again.

**Comment by avanwouwe** [comment]
I am on 4.7.3 and my agent running on Windows 11 PC is has this issue.

The fix by vikman90 was merged into the 4.8.2 branch, which does not seem to exist anymore. Searching through the code, the change no longer appears in any branches, except in "master". Is that normal?

The ticket is now tagged for 4.9.0 but that seems quite far away? Certainly for something that breaks osquery integration on Windows?

I guess I must be missing something, for this not to be making more noise?

**Comment by vikman90** [comment]
@avanwouwe, we decided to merge fixes that we were going to deliver in 4.8.x to 4.9.0, in order to boost up this one.

I confirm that the fix is in branch 4.9.0:
https://github.com/wazuh/wazuh/blob/3125655b9e7f4db88b1587a1e2cc840c31880bbb/src/shared/file_op.c#L3347

This release is planned for Q3.

## Workaround

As a workaround, I can propose to temporarily avoid any configuration that makes the module parse the osquery configuration (which is in JSON format):
- `add_labels`
- `pack`

Sorry for the inconveniences.

Best regards.
